libcalico-go/lib/health/health.go

```diff
@@ -249,29 +255,23 @@ func (aggregator *HealthAggregator) Summary() *HealthReport {
 	// Now for each reporter...
 	for _, reporter := range aggregator.reporters {
 		log.WithField("reporter", reporter).Debug("Checking state of reporter")
-		livenessStr := "-"
-		if reporter.HasLivenessProblem() {
-			log.WithField("name", reporter.name).Warn("Reporter is not live.")
-			summary.Live = false
-			if reporter.TimedOut() {
-				livenessStr = "timed out"
+		live, livenessStr := reporter.liveness()
+		if !live {
+			if aggregator.previouslyLive {
+				log.WithField("name", reporter.name).Warnf("Reporter is not live: %v.", livenessStr)
 			} else {
-				livenessStr = "reporting non-live"
+				log.WithField("name", reporter.name).Infof("Reporter is not live: %v.", livenessStr)
```

Comment:
For liveness, would it make more sense to assume that a component is live at start of day (until first timeout)?  Letting liveness time out is really the only sensible way to go non-live (in any other case, why set non-live instead of panicing or doign a graceful shutdown?)

Reply:
We do have that start of day assumption for liveness here: https://github.com/projectcalico/calico/blob/master/libcalico-go/lib/health/health.go#L150

I suppose that means we could remove the `previouslyLive` field, because it should always be true.  I'll do that.

You might also be suggesting: why would we support a reporter later being able to say `Live: false` ?  I don't think there are any reporters (in our repos) that can ever do that, so I agree that we could rip that out too.